install/kubernetes/cilium/templates/spire/server/statefulset.yaml

```diff
@@ -0,0 +1,80 @@
+{{- if and .Values.auth.mTLS.spire.enabled .Values.auth.mTLS.spire.install.enabled -}}
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: spire-server
+  namespace: {{ .Values.auth.mTLS.spire.install.namespace }}
+  labels:
+    app: spire-server
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: spire-server
+  serviceName: spire-server
+  template:
+    metadata:
+      namespace: {{ .Values.auth.mTLS.spire.install.namespace }}
```

Comment:
This shouldn't be necessary. Pods are always created in the same NS as the controller